src/app/models/form.server.model.ts

```diff
@@ -208,7 +211,62 @@ const compileFormModel = (db: Mongoose): IFormModel => {
             'Check that your form is MyInfo-authenticated, is an email mode form and has 30 or fewer MyInfo fields.',
         },
       },
-      form_logics: [LogicSchema],
+      form_logics: {
+        type: [LogicSchema],
+        validate: {
+          validator(this: IFormSchema, v: ILogicSchema[]) {
+            /**
+             * A validatable condition is incomplete if there is a possibility
+             * that its fieldType is null, which is a sign that a condition's
+             * field property references a non-existent form_field.
+             */
+            type IncompleteValidatableCondition = {
+              state: LogicConditionState
+              fieldType?: BasicField
+            }
+
+            /**
+             * A condition object is said to be validatable if it contains the two
+             * necessary for validation: fieldType and state
+             */
+            type ValidatableCondition = IncompleteValidatableCondition & {
+              fieldType: BasicField
+            }
+
+            const isConditionReferencesExistingField = (
+              condition: IncompleteValidatableCondition,
+            ): condition is ValidatableCondition => !!condition.fieldType
+
+            const conditions = v.flatMap((logic) => {
+              return logic.conditions.map<IncompleteValidatableCondition>(
+                (condition) => {
+                  const {
+                    field,
+                    state,
+                  }: { field: ObjectId | string; state: LogicConditionState } =
+                    condition
+                  return {
+                    state,
+                    fieldType: this.form_fields?.find(
+                      (f: IFieldSchema) => String(f._id) === String(field),
+                    )?.fieldType,
+                  }
+                },
+              )
+            })
+
+            return conditions.every((condition) => {
+              // Conditions don't necessarily need to reference existing fields
```

Comment:
would be good if we can add a bit more explanation here to explain that we allow logic referencing non-existent fields to be saved, so we don't make life difficult for form admins when they delete fields

Reply:
added more documentation, lmk what you think!

Reply:
lgtm!